docs/tutorials/mobile/superres.md

```diff
@@ -0,0 +1,606 @@
+   * `OnnxTensor.createTensor()` is used to create a tensor made up of the input image bytes, suitable as input to the model
+   * `OnnxJavaType.UINT8` is the data type of the input tensor
+   * `OrtSession.run()` run the inference (prediction) on the model to get the output upscaled image
+
+   ```kotlin
+   internal class SuperResPerformer(
+    private val ortSession: OrtSession
+   ) {
+
+       fun upscale(inputStream: InputStream, ortEnv: OrtEnvironment): Result {
+           var result = Result()
+
+           // Step 1: convert image into byte array (raw image bytes)
+           val rawImageBytes = inputStream.readBytes()
+
+           // Step 2: get the shape of the byte array and make ort tensor
+           val shape = longArrayOf(rawImageBytes.size.toLong())
+
+           inputTensor.use {
+               // Step 3: call ort inferenceSession run
+               val output = ortSession.run(Collections.singletonMap("image", inputTensor))
+
+               // Step 4: output analysis
+               output.use {
+                   val rawOutput = (output?.get(0)?.value) as ByteArray
+                   val outputImageBitmap =
+                       byteArrayToBitmap(rawOutput)
+
+                   // Step 5: set output result
+                   result.outputBitmap = outputImageBitmap
+               }
+           }
+           return result
+       }
+
+       private fun byteArrayToBitmap(data: ByteArray): Bitmap {
+           return BitmapFactory.decodeByteArray(data, 0, data.size)
+       }
+   }
+   ```
+
+### Build and run the app
+
+Within Android studio:
+
+* Select Build -> Make Project
+* Run -> app
+
+The app runs in the device emulator. Connect to your Android device to run the app on device.
+
+## iOS app
+
+### Pre-requisites
+
+* Install Xcode 13.0 and above (preferably latest version)
+* A valid Apple Developer ID
+* An iOS device or iOS simulator
+* Xcode command line tools `xcode-select --install`
+* CocoaPods `sudo gem install cocoapods`
+
+### Sample code
+
+You can find full [source code for the iOS super resolution app](https://github.com/microsoft/onnxruntime-inference-examples/tree/main/mobile/examples/super_resolution/ios) in GitHub.
+
+To run the app from source code:
+
+1. Clone the onnxruntime-inference-examples repo
+
+   ```bash
+   git clone https://github.com/microsoft/onnxruntime-inference-examples
+   cd onnxruntime-inference-examples/mobile/examples/super_resolution/ios
+   ```
+
+2. Install required pod files
+
+   ```bash
+   pod install
+   ```
+
+3. Open the generated `ORTSuperResolution.xcworkspace` file in XCode
+
+   Select your development team
+
+4. Run the application
+
+   Connect your iOS device or simulator, build and run the app
+
+   Click the `Perform Super Resolution` button to see the app in action
+
+To develop the app, step by step, follow the following sections.
+
+### Code from scratch
+
+#### Create project
+
+Create a new project in XCode
```

Comment:
Not sure if it's useful info, fyi: 

for iOS, we use the `APP` template, and there are two options to choose for interface a) SwiftUI b) Storyboard.

For the current basic super resolution we started with creating the app based on SwiftUI interface, which will create a ContentView based application. (slightly simpler in terms of app structure.) https://developer.apple.com/tutorials/swiftui/creating-and-combining-views

Storyboard option will create a ViewController based application, which is also workable and would be good to visualize different layers of views (if we have more complex UI in app,  say introducing camera app functionality, we may use that as well.)